edx_lint/pylint/right_assert_check.py

```diff
@@ -51,39 +85,21 @@ def visit_callfunc(self, node):
             return
 
         first_arg = node.args[0]
-        if not isinstance(first_arg, astroid.Compare):
-            # Not a comparison, so this is probably ok:
-            return
-
-        if first_arg.ops[0][0] in ["==", "!="]:
-            # An assertTrue/False with a compare should be assertEqual:
-            self.add_message(
-                self.MESSAGE_ID,
-                args="%s(%s) should be assertEqual or assertNotEqual" % (
-                    node.func.attrname,
-                    first_arg.as_string(),
-                ),
-                node=node,
-            )
-
-        elif first_arg.ops[0][0] in ["in", "not in"]:
-            # An assertTrue/False with an in statement should be assertIn:
-            self.add_message(
-                self.MESSAGE_ID,
-                args="%s(%s) should be assertIn or assertNotIn" % (
-                    node.func.attrname,
-                    first_arg.as_string(),
-                ),
-                node=node,
-            )
-
-        elif "<" in first_arg.ops[0][0] or ">" in first_arg.ops[0][0]:
-            # An assertTrue/False with a comparison should be assertGreater or assertLess:
+        existing_code = "%s(%s)" % (node.func.attrname, first_arg.as_string())
+
+        if isinstance(first_arg, astroid.Compare):
+            compare = first_arg.ops[0][0]
+            right = first_arg.ops[0][1]
```

Comment:
I wondered about that.  In fact, because of chained comparisons, the structure of a Compare node is that `left` is an expression, and then `ops` is a list of pairs, (compare_op, expression).  So `a == b == c <= d` is expressed as `(left=a, ops=[("==", b), ("==", c), ("<=", d)])`.  We don't check here if there is more than one comparison either.